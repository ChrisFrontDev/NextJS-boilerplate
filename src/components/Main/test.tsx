import { render, screen } from '@testing-library/react';
import Main from './';

describe('test <Main /> component', () => {
  const { container } = render(<Main />);

  it('should render heading', () => {
    expect(
      screen.getByRole('heading', { name: /React AvanÃ§ado/i })
    ).toBeInTheDocument();
  });

  it('should render colors correctly', () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
  });

  expect(container.firstChild).toMatchSnapshot();
});
