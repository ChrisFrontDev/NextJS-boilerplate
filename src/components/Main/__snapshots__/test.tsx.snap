// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 elJNpC"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="styles__Logo-dzx1ao-1 dUetyB"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2 edokyf"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 gfCskk"
  >
    TypeScript.ReactJs,NextJs e Styled Components
  </h2>
  <img
    alt="Um desenvovedor em frente ao computador escrevendo o código"
    class="styles__Illustrator-dzx1ao-4 oGqLY"
    src="/img/hero-illustration.svg"
  />
</main>
`;
